modules/MouseModels.py

```diff
@@ -832,20 +851,39 @@ def process_all(self, update_cache=False, write2cloud=False):
         #bar.update()
         return
 
-def main():
 
-    parser = optparse.OptionParser()
-    parser.add_option("-u", '--update-cache', action="store_true", dest="update_cache", default=False)
-    parser.add_option("-w", '--write-evidence-to-cloud', action="store_true", dest="write2cloud", default=False)
-    options, args = parser.parse_args()
+def main():
 
-    # -u and -w options are mutually exclusuve
-    if options.update_cache and options.write2cloud:
-        parser.error("Options -u and -w are mutually exclusive, please use one only")
+    parser = argparse.ArgumentParser(description='Evidence parser for animal models sources from PhenoDigm')
+
+    # Two of the flags are mutually exclusive:
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-u", '--update-cache', help='To use cached data instead of fetching from IUPMC solr', 
+        action="store_true", dest="update_cache", default=False)
+    group.add_argument("-w", '--write-evidence-to-cloud', help='If the file is expected ot be uploaded to the cloud right a way',
+        action="store_true", dest="write2cloud", default=False)
+
+    # log file is optional:
+    parser.add_argument("-l", '--log-file', help='Optional filename for logfile.', required=False)
+    args = parser.parse_args()
+
+    # Initialize logger:
+    # Initialize logger based on the provided logfile. 
+    # If no logfile is specified, logs are written to stderr 
+    logging.basicConfig(
```

Comment:
Could this logger conflict with the one implemented within the class?

Reply:
I did a bit of a cleanup in the logging. We'll see how it works.